x/auth/legacy/v043/store_test.go

```diff
@@ -0,0 +1,593 @@
+			vestingAccount, ok := savedAccount.(exported.VestingAccount)
+			require.True(t, ok)
+			require.NoError(t, introduceTrackingBug(ctx, vestingAccount, app))
+
+			newData, err := v043.MigrateStore(ctx, app.AccountKeeper.GetAllAccounts(ctx), app.GRPCQueryRouter())
+			require.NoError(t, err)
+
+			for _, a := range newData {
+				app.AccountKeeper.SetAccount(ctx, a)
+			}
+
+			var expVested sdk.Coins
+			var expFree sdk.Coins
+
+			if tc.expVested != 0 {
+				expVested = sdk.NewCoins(sdk.NewCoin(app.StakingKeeper.BondDenom(ctx), sdk.NewInt(tc.expVested)))
+			}
+
+			if tc.expFree != 0 {
+				expFree = sdk.NewCoins(sdk.NewCoin(app.StakingKeeper.BondDenom(ctx), sdk.NewInt(tc.expFree)))
+			}
+
+			trackingCorrected(
+				ctx,
+				t,
+				app.AccountKeeper,
+				savedAccount.GetAddress(),
+				expVested,
+				expFree,
+			)
+		})
+	}
+
+}
+
+func trackingCorrected(ctx sdk.Context, t *testing.T, ak authkeeper.AccountKeeper, addr sdk.AccAddress, expDelVesting sdk.Coins, expDelFree sdk.Coins) {
+	t.Helper()
+	baseAccount := ak.GetAccount(ctx, addr)
+	vDA, ok := baseAccount.(exported.VestingAccount)
+	require.True(t, ok)
+
+	vestedOk := expDelVesting.IsEqual(vDA.GetDelegatedVesting())
+	freeOk := expDelFree.IsEqual(vDA.GetDelegatedFree())
+	require.True(t, vestedOk, vDA.GetDelegatedVesting().String())
+	require.True(t, freeOk, vDA.GetDelegatedFree().String())
+}
+
+func introduceTrackingBug(ctx sdk.Context, vesting exported.VestingAccount, app *simapp.SimApp) error {
+	switch t := vesting.(type) {
+	case *types.DelayedVestingAccount:
+		t.DelegatedFree = nil
+		t.DelegatedVesting = nil
+		app.AccountKeeper.SetAccount(ctx, t)
+	case *types.ContinuousVestingAccount:
+		t.DelegatedFree = nil
+		t.DelegatedVesting = nil
+		app.AccountKeeper.SetAccount(ctx, t)
+	case *types.PeriodicVestingAccount:
+		t.DelegatedFree = nil
+		t.DelegatedVesting = nil
+		app.AccountKeeper.SetAccount(ctx, t)
+	default:
+		return fmt.Errorf("expected vesting account, found %t", t)
+	}
+
+	return nil
+}
+
+func createValidator(t *testing.T, ctx sdk.Context, app *simapp.SimApp, powers int64) (sdk.AccAddress, sdk.ValAddress) {
+	addrs := simapp.AddTestAddrsIncremental(app, ctx, 1, sdk.NewInt(30000000))
+	valAddrs := simapp.ConvertAddrsToValAddrs(addrs)
+	pks := simapp.CreateTestPubKeys(1)
+	cdc := simapp.MakeTestEncodingConfig().Marshaler
+
+	app.StakingKeeper = stakingkeeper.NewKeeper(
+		cdc,
+		app.GetKey(stakingtypes.StoreKey),
+		app.AccountKeeper,
+		app.BankKeeper,
+		app.GetSubspace(stakingtypes.ModuleName),
+	)
+
+	val1, err := stakingtypes.NewValidator(valAddrs[0], pks[0], stakingtypes.Description{})
+	require.NoError(t, err)
+
+	app.StakingKeeper.SetValidator(ctx, val1)
+	require.NoError(t, app.StakingKeeper.SetValidatorByConsAddr(ctx, val1))
+	app.StakingKeeper.SetNewValidatorByPowerIndex(ctx, val1)
+
+	_, _ = app.StakingKeeper.Delegate(ctx, addrs[0], sdk.TokensFromConsensusPower(powers), stakingtypes.Unbonded, val1, true)
```

Comment:
All errors should be checked.

Reply:
Checked, and uncovered an issue in test preparation phase.

Thanks!